File: augur-engine/src/config/horizons.py
"""Horizon date computation and section labels."""

from __future__ import annotations

from datetime import datetime, timedelta
from dateutil.relativedelta import relativedelta


def compute_fictive_date(horizon: str, anchor: datetime | None = None) -> str:
    """Compute the fictive prediction date for a given horizon."""
    d = anchor or datetime.utcnow()
    if horizon == "tomorrow":
        d += timedelta(days=1)
    elif horizon == "soon":
        d += relativedelta(months=1)
    elif horizon == "future":
        d += relativedelta(years=1)
    return d.strftime("%Y-%m-%d")


def today_key() -> str:
    return datetime.utcnow().strftime("%Y-%m-%d")


SECTION_LABELS = {
    "en": {
        "signal": "The Signal",
        "extrapolation": "The Extrapolation",
        "in_the_works": "In The Works",
        "sources": "Sources",
        "sentiment": "The Augur's Sentiment",
        "foreseen_for": "Foreseen for",
        "divined": "Divined",
    },
    "de": {
        "signal": "Das Signal",
        "extrapolation": "Die Extrapolation",
        "in_the_works": "In Arbeit",
        "sources": "Quellen",
        "sentiment": "Die Einschätzung des Augur",
        "foreseen_for": "Vorhergesagt für",
        "divined": "Erstellt",
    },
}

File: augur-engine/src/config/horizons.ts
import type { HorizonKey } from './types.js';

/** Compute the fictive date for a given horizon from an anchor date */
export function computeFictiveDate(horizon: HorizonKey, anchor: Date = new Date()): string {
  const d = new Date(anchor);

  switch (horizon) {
    case 'tomorrow':
      d.setDate(d.getDate() + 1);
      break;
    case 'soon':
      d.setMonth(d.getMonth() + 1);
      break;
    case 'future':
      d.setFullYear(d.getFullYear() + 1);
      break;
  }

  return d.toISOString().slice(0, 10); // YYYY-MM-DD
}

/** Get today's date key */
export function todayKey(): string {
  return new Date().toISOString().slice(0, 10);
}

/** Section labels per locale */
export const SECTION_LABELS = {
  en: {
    signal: 'The Signal',
    extrapolation: 'The Extrapolation',
    inTheWorks: 'In The Works',
    sources: 'Sources',
    sentiment: "The Augur's Sentiment",
    foreseenFor: 'Foreseen for',
    divined: 'Divined',
  },
  de: {
    signal: 'Das Signal',
    extrapolation: 'Die Extrapolation',
    inTheWorks: 'In Arbeit',
    sources: 'Quellen',
    sentiment: 'Die Einschätzung des Augur',
    foreseenFor: 'Vorhergesagt für',
    divined: 'Erstellt',
  },
} as const;
